subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r43/ToolingApiDeprecatedProviderCrossVersionSpec.groovy

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.r43
+
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+
+@ToolingApiVersion("current")
+class ToolingApiDeprecatedProviderCrossVersionSpec extends ToolingApiVersionSupport {
+    @TargetGradleVersion(">=1.2 <2.6")
+    def "deprecation warning in build execution for >=1.2 and <2.6 providers"() {
+        expect:
+        buildViaScript().count(providerDeprecationMessage(targetDist.version.version)) == 1
```

Comment:
❌ We should test this like all other cases, not with the indirection through a script. I guess you did this because it was the only way to see the deprecation message. But that just means a user would never see it. So instead of adjusting the test, we need to fix the implementation. 

The message is currently logged too early and only to stdOut. Instead, it should only be logged when an actual request is made and it should be written to the output stream that was passed with that request.